modules/imgproc/src/templmatch.cpp

```diff
@@ -740,6 +747,16 @@ void crossCorr( const Mat& img, const Mat& _templ, Mat& corr,
     }
     borderType |= BORDER_ISOLATED;
 
+    bool useHalDft = tileCount > 1;
+    hal::DftContext cF, cR;
+    if (useHalDft)
```

Comment:
I would simplify it and use HAL all the time here; the case tileCount == 1 is not that important